src/config.ts

```diff
@@ -25,23 +26,31 @@ const defaultConfig: Config = {
 
 let config: Config | undefined
 
-export function getConfig() {
+export async function getConfig(): Promise<Config> {
   if (!config) {
-    if (!fs.existsSync(rcPath))
+    const result = await findUp('package.json') || ''
+    let packageManager = ''
+    if (result)
+      packageManager = JSON.parse(fs.readFileSync(result, 'utf8')).packageManager ?? ''
+    const [, agent] = packageManager.match(new RegExp(`^(${Object.values(LOCKS).join('|')})@.*?$`)) || []
```

Comment:
I guess we might need to check the agent is in `npm` `yarn` `pnpm` otherwise emit a warning of unsupported package manager and fallback to the original behaviour

Reply:
```suggestion
36L has check
```